nilearn/datasets/func.py

```diff
@@ -1087,6 +1087,17 @@ def fetch_abide_pcp(
     HANDEDNESS_SCORE : integer in [-100, 100], optional
         Positive = Right, Negative = Left, 0 = Ambidextrous.
 
+    Returns
+    -------
+    data : :class:`sklearn.utils.Bunch`
+        Dictionary-like object, keys are:
+
+            - 'description': :obj:`str`, description of the dataset.
+            - 'phenotypic': :obj:`pandas.DataFrame`, a pandas DataFrame
+              containing phenotypic information for each subject.
+            - 'func_preproc': :obj:`numpy.ndarray`, a numpy array containing
+              paths to preprocessed functional MRI data in NIfTI format.
```

Comment:
Was checking how this fetcher works and this may need to be adapted.

For example if I ask for other derivatives, the bunch I get in return has more keys.

For example:

```python
from nilearn.datasets import fetch_abide_pcp

results = fetch_abide_pcp(
    n_subjects=1,
    pipeline="cpac",
    derivatives=["func_preproc", "func_mean"],
    verbose=1,
)

print(results.keys())
```

Gives:

```python
dict_keys(['description', 'phenotypic', 'func_preproc', 'func_mean'])
```

Also the return type would be

- :obj:`list` of :obj:`numpy.ndarray` for derivative that start with "rois" and :obj:`list` of :obj:`str` to the path to the nifti images otherwise.

Reply:
Hmm interesting, I'll look into it

Reply:
yeah sorry some of the "fetchers" can return a wide variety of outputs: makes writing the doc a bit more involved

Reply:
Would it be an idea to document the default return and add a note that additional/different dict outputs are possible when setting other values in `derivatives`?

Reply:
yes that sounds like a good idea!